feature-libs/organization/administration/core/guards/admin.guard.ts

```diff
@@ -15,17 +15,33 @@ import {
 import { UserAccountFacade } from '@spartacus/user/account/root';
 import { Observable } from 'rxjs';
 import { filter, map, pluck } from 'rxjs/operators';
+import { B2BUserService } from '../services';
+import { Location } from '@angular/common';
 
 @Injectable({
   providedIn: 'root',
 })
-export class AdminGuard implements CanActivate {
+export class AdminGuard implements CanActivate, CanMatch {
   constructor(
     protected userAccountFacade: UserAccountFacade,
     protected routingService: RoutingService,
-    protected globalMessageService: GlobalMessageService
+    protected globalMessageService: GlobalMessageService,
+    protected b2bUserService: B2BUserService,
+    protected location: Location
   ) {}
 
+  canMatch(): boolean {
```

Comment:
I suggest not to enlarge the scope of the `AdminGuard`, which is now responsible only for protecting pages accessible only for the Admin's role. Instead I suggest to create a brand new guard for your custom use case.